libs/flexIPC/fleXdIPCCommon/src/FleXdIPCMsg.cpp

```diff
@@ -0,0 +1,466 @@
+            }
+
+            FleXdIPCMsg::~FleXdIPCMsg()
+            {
+                if (m_additionalHeader != nullptr)
+                {
+                    delete m_additionalHeader;
+                }
+            }
+
+            FleXdIPCMsg::FleXdIPCMsg(FleXdIPCMsg&& other)
+            : m_payload(std::move(other.m_payload)),
+              m_headerParamType(other.m_headerParamType),
+              m_headerParam(other.m_headerParam),
+              m_additionalHeader(other.m_additionalHeader),
+              m_valid(other.m_valid)
+            {
+                other.m_headerParamType = 0;
+                other.m_headerParam = 0;
+                other.m_additionalHeader = nullptr;
+                other.m_valid = false;
+            }
+
+            FleXdIPCMsg& FleXdIPCMsg::operator=(FleXdIPCMsg&& other)
+            {
+                if (this != &other)
+                {
+                    m_payload.clear();
+                    if (m_additionalHeader != nullptr) delete m_additionalHeader;
+
+                    m_payload = std::move(other.m_payload);
+                    m_headerParamType = other.m_headerParamType;
+                    m_headerParam = other.m_headerParam;
+                    m_additionalHeader = other.m_additionalHeader;
+                    m_valid = other.m_valid;
+
+                    other.m_additionalHeader = nullptr;
+                    other.m_valid = false;
+                }
+                return *this;
+            }
+
+            const bool FleXdIPCMsg::getHeaderParamType() const
+            {
+                return m_headerParamType;
+            }
+
+            const uint8_t FleXdIPCMsg::getHeaderParam() const
+            {
+                if (m_headerParamType && m_additionalHeader != nullptr) return m_additionalHeader->getMask();
+                return m_headerParam;
+            }
+
+            void FleXdIPCMsg::setHeaderParam(uint8_t headerParam)
+            {
+                if (!m_headerParamType) m_headerParam = headerParam;
+            }
+
+            void FleXdIPCMsg::resetHeaderParam()
+            {
+                m_headerParam = 0;
+                if (m_headerParamType) deleteAdditionalHeader();
+            }
+
+            const uint16_t FleXdIPCMsg::getMsgSize() const
+            {
+                return IPC_MSG_HEADER_SIZE + m_payload.size() + (m_additionalHeader == nullptr ? 0 : m_additionalHeader->getSize());
+            }
+
+            FleXdIPCAdtHdr* FleXdIPCMsg::createAdditionalHeader()
+            {
+                if (m_headerParamType && m_additionalHeader == nullptr && m_headerParam != IPC_MSG_HEADER_IN_PAYLOAD_FLAG)
+                {
+                    m_additionalHeader = new FleXdIPCAdtHdr();
+                }
+                return m_additionalHeader;
+            }
+
+            void FleXdIPCMsg::deleteAdditionalHeader()
+            {
+                if (m_additionalHeader != nullptr)
```

Comment:
Ditto 4. 'if (m_next != nullptr)' is fairly useless since standard defines the behaviour of 'delete' well when 'nullptr' is supplied.